packages/flutter/lib/src/rendering/object.dart

```diff
@@ -3643,7 +3661,7 @@ abstract class RenderObject with DiagnosticableTreeMixin implements HitTestTarge
         mayProduceSiblingNodes = false;
       }
 
-      node = node.parent!;
+      node = node.semanticsParent!;
```

Comment:
Looks like this line is the only reason that `RenderObject.semanticsParent` is needed. Is it possible to change the implementation of `markNeedsSemanticsUpdate` to a recursive one so we can override `_RenderDeferredLayoutBox.markNeedsSemanticsUpdate` to propagate the call to its surrogate parent? So that we don't have to change the interface of `RenderObject`.

Reply:
I cant think of a clean way to do this with the `mayProduceSiblingNodes`. The requirement is the following:

If a node will produce sibling nodes, it must mark two semantics boundaries above to be dirty

If we were to do this, we have to add additional flag either on renderobject or additional parameter to the markNeedsSemanticsUpdate([int howManyBoundaries]).

Reply:
The semanticsparent is also used for some look up in _getSemanticsForParent.

Reply:
> The semanticsparent is also used for some look up in _getSemanticsForParent.

The call from `_getSemanticsForParent` only wants to check if the parent is null it seems (so it doesn't matter for OverlayPortal). 

> If a node will produce sibling nodes, it must mark two semantics boundaries above to be dirty

Ah ok that makes sense. (But at the same time that feels a bit weird, not familiar with semantics tree, but why is the closest semantics boundary invalidated if `this` has child configurations, but it's fine if `parent` has child configurations?)

Reply:
if the parent is not markNeedsSemanticsUpdate, it won't change its sibling node that get pass on two level up.

If the parent it self is markNeedsSemanticsUpdate it is not the current node's job to mark two level above the parent dirty. It will be when the parent's markNeedsSemanticsUpdate call to mark two level above.